### PR TITLE
✨ Use helm chart releaser and generate notes on each release

### LINKED ISSUE #23: Automate extension build
#### Story

The extension should be built automatically via GitHub workflows.

#### Description

Existing Rancher extensions [implement a series of GitHub Actions](https://github.com/kubewarden/ui/actions/workflows/build-extension.yml) that automatically build the extension package. Based off these workflows, automate building and packaging the CAPI extension.

### PR DESCRIPTION
**Pull request by Danil-Grigorev**
kind/feature

**What this PR does / why we need it**:

This PR is implementing support for publishing charts with [chart-releaser](https://github.com/helm/chart-releaser) tool. Additionally adds auto release note generation for each release based on CAPI [release/note](https://github.com/kubernetes-sigs/cluster-api/blob/main/hack/tools/release/notes.go) tool.

Here is an example release created by this workflow: https://github.com/Danil-Grigorev/rancher-turtles/releases/tag/rancher-turtles-1.2.20, might be good to have labeling on PRs similar to CAPI operator or CAPI. 

Additionally it removes `rancher-turtles-components.yaml` file created with each release, as there is no need for that in rancher-turtles opposed to CAPI operator.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #23

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: hack/make-release-notes.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+set -e
+set -o errexit
+set -o pipefail
+
+TOOLS_FOLDER=${1:-../hack/tools/bin}
+
+mkdir $TOOLS_FOLDER/cluster-api
+cd $TOOLS_FOLDER/cluster-api
```
Comment (alexander-demicev) [inline hack/make-release-notes.sh:10]: I think this script has to be adjusted for rancher turles

File: Makefile [hunk 2]
```diff
@@ -356,6 +364,9 @@ $(GOLANGCI_LINT): # Download and install golangci-lint
 		-b $(TOOLS_BIN_DIR) \
 		$(GOLANGCI_LINT_VER)
 
+$(NOTES): # Download and install note generator from cluster-api commit
```
Comment (furkatgofurov7) [inline Makefile:367]: we need to remove capi notes, perhaps?

**Review by furkatgofurov7 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (Danil-Grigorev) [reply Makefile:367]: None of the created binaries are getting removed by any of the makefile tasks, why should it be?

File: hack/make-release-notes.sh (hunk 1 above)
Comment (Danil-Grigorev) [reply hack/make-release-notes.sh:10]: I'm not sure I understand. This script was created for rancher turtles specifically, what I missing here to be adjusted?

**Comment by Danil-Grigorev** [comment]
One gap I'm seeing now, is that the image field is not configurable by helm in any way. So this helm chart is not tailored to specific architecture as it should probably be based on the docker images built, it just gets the amd64 image due to CI environment. Not sure though if it was addressed in the initial implementation of the make release. But we could populate values.yaml later, and add templating for the created resource, similar to https://github.com/kubernetes-sigs/cluster-api-operator/pull/199. And we could remove kustomize by incremental refactoring I guess.

**Review by alexander-demicev — APPROVED** [review]
In the future, we should talk with CAPI folks about converting the release note tool to a github action, with some changes it can be reused by CAPI providers.

File: hack/make-release-notes.sh (hunk 1 above)
Comment (alexander-demicev) [reply hack/make-release-notes.sh:10]: Ok, I see there is no way to import the release note tool here

File: Makefile (hunk 2 above)
Comment (alexander-demicev) [reply Makefile:368]: @furkatgofurov7 There is no nice way to import CAPI release tool here, all methods are private. The only option is to build it from CAPI repo. We can talk with CAPI folks about converting it to a github action.

**Review by furkatgofurov7 — APPROVED** [review]

**Pull request merged**